### DISCUSSION TITLE
Guidance for testing ListRecords

### DISCUSSION THREAD

**Discussion by titantwentyone** [filamentphp/filament#1825] [Help]
I can see there is some guidance on how to test Edit and Create pages and functions for admin but couldn't see anything to test that the ListRecords page returns the correct data. The only thing I could find was verifiying that the link renders the ListRecords page [here](https://github.com/laravel-filament/filament/blob/2.x/tests/src/Admin/Resources/Pages/ListRecordsTest.php).

Would also be good, if possible, to get some guidance on how to test any filters. If I can get some pointers, I'd gladly amend the docs.

**Comment by danharrin** [comment]
Honestly I always get stuck on how to test lists of records without just using `assertSee()` of various pieces of data. Any suggestions from the community are appreciated.

**Comment by titantwentyone** [comment]
Yeh. `assertSee` always feels a bit flimsy sometimes. Am doing this for one particular resource though:

```
$campevent = CampEvent::factory()->create();
$this->assertEquals(CampEvent::withoutGlobalScopes(), CampEventResource::getEloquentQuery());
```

I'm only doing this since I changed my resource query to exclude deleted items and I'm in two minds as to whether this is actually neded to be honest.

```
public static function getEloquentQuery(): Builder
{
    return parent::getEloquentQuery()->withoutGlobalScopes();
}
```
Currently looking to see how I can effectively test `Filters` and any `Columns` that have things like `getStateUsing` etc.

**Comment by danharrin** [comment]
https://filamentphp.com/docs/2.x/tables/testing
